.github/workflows/e2e.yml

```
@@ -0,0 +1,84 @@
+name: e2e suite
+
+on:
+  workflow_dispatch:
+  schedule:
+    # At 2:00 AM UTC on Monday, Wednesday, and Friday
+    - cron: "00 2 * * 1,3,5"
+  pull_request:
+    branches:
+      - 'main'
+
+env:
+  BENCH_PATH: ./queries-results.json
+  DATABASE_URL: "postgres://saleor:saleor@postgres:5432/saleor"
+  SECRET_KEY: ci-test
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container: python:3.9
+
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_PASSWORD: saleor
+          POSTGRES_USER: saleor
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install system dependencies
+        run: |
+          apt-get install -y libpq-dev
+
+      - name: Install and configure poetry
+        run: |
+          pip install poetry==1.7.0
+
+      - name: Cache the venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v3
+        with:
+          path: ~/.venv
+          key: ${{ runner.os }}-venv-${{ hashFiles('**/poetry.lock') }}
+
+      - name: Install dependencies
+        run: |
+          poetry env use python3.9
+          poetry install --no-root
+        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+
+      - name: Add Python Virtual Environment to PATH
+        run: |
+          # Note: requires `poetry env use` to be ran, otherwise poetry
+          # may not be able to find which virtual environment is in use.
+          # Ticket: https://github.com/python-poetry/poetry/issues/7190
+          echo "$(poetry env info -p)"/bin >> $GITHUB_PATH
+
+      - uses: actions/cache@v3
+        with:
+          path: ~/.cache/pre-commit
+          key: ${{ runner.os }}-pre-commit-${{ hashFiles('**/.pre-commit-config.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-pre-commit-
+
```

pre-commit isn't used in that workflow thus it's not needed
```suggestion
          echo "$(poetry env info -p)"/bin >> $GITHUB_PATH

```